cmake/Codegen.cmake

```diff
@@ -122,6 +122,22 @@ if (INTERN_BUILD_ATEN_OPS)
   list(LENGTH CPU_CAPABILITY_NAMES NUM_CPU_CAPABILITY_NAMES)
   math(EXPR NUM_CPU_CAPABILITY_NAMES "${NUM_CPU_CAPABILITY_NAMES}-1")
 
+  IF(CXX_AVX512_FOUND)
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DHAVE_AVX512_CPU_DEFINITION")
+
+    check_cxx_compiler_flag("-mno-avx256-split-unaligned-load -mno-avx256-split-unaligned-store" COMPILER_SUPPORTS_NO_AVX256_SPLIT)
+    IF(COMPILER_SUPPORTS_NO_AVX256_SPLIT)
+      SET(CPU_NO_AVX256_SPLIT_FLAGS "-mno-avx256-split-unaligned-load -mno-avx256-split-unaligned-store")
+    ENDIF(COMPILER_SUPPORTS_NO_AVX256_SPLIT)
+
+    LIST(APPEND CPU_CAPABILITY_NAMES "AVX512")
+    IF(MSVC)
+      LIST(APPEND CPU_CAPABILITY_FLAGS "${OPT_FLAG}/arch:AVX512")
```

Comment:
Is this flag valid? I don't see it in the docs https://docs.microsoft.com/en-us/cpp/build/reference/arch-x64?view=vs-2019

Reply:
Nvm, I see an issue at the bottom of that page saying it's undocumented